network/p2p/gossip/set_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
```

Comment:
I saw you added the locks, but I don't think these are tested anywhere. 

Something like `func TestSetWithBloomFilter_Concurrent`?